### PR TITLE
Remove dependence on eventdispatcher [Delivers JW-1254]

### PR DESCRIPTION
**Pull request by pajong**
Removed eventdispatcher and updated providers and model accordingly

### CODE REVIEW

File: src/js/providers/html5.js [hunk 1]
```diff
@@ -59,8 +59,19 @@ define([
...
+                callback(input);
+            });
+        };
+
+        _.extend(this, Events);
+
+        // legacy support
+        this.addEventListener = this.on;
+        this.sendEvent = this.trigger;
+        this.resetEventListeners = this.removeEventListener = this.off;
```
Comment (donato) [inline src/js/providers/html5.js]: Let these be gone!

**Comment by donato** [comment]
Good start.

Now from here, delete all the "legacy" support code. Remove "addGlobalEventListener" from everywhere. The places that used to use "global event" are where you will move the "on('all',...)" code.

File: src/js/controller/instream-html5.js [hunk 2]
```diff
@@ -103,7 +103,6 @@ define([
 
                 provider.resetEventListeners();
 
-                provider.addGlobalListener(_forward);
```
Comment (donato) [inline src/js/controller/instream-html5.js]: I would expect this to be replaced by something else, not just removed. Will this work?

File: src/js/controller/instream-html5.js [hunk 3]
```diff
@@ -101,12 +101,11 @@ define([
                     return;
                 }
 
-                provider.resetEventListeners();
+                provider.off();
 
-                provider.addGlobalListener(_forward);
```
Comment (donato) [inline src/js/controller/instream-html5.js]: I would expect to see ```provider.on('all', ...)``` here, is it not needed?

File: src/js/providers/flash.js [hunk 4]
```diff
@@ -348,18 +344,10 @@ define([
...
                     _item = null;
-                    _eventDispatcher.resetEventListeners();
-                    _eventDispatcher = null;
+                    this.off();
                 }
         });
 
-        // Overwrite the event dispatchers to block on certain occasions
-        this.sendEvent = function() {
-            if (!_attached) {
```
Comment (robwalch) [inline src/js/providers/flash.js]: The functionality of this `if (!_attached) {` statement is important, and it looks like it's been removed.

File: src/js/providers/html5.js [hunk 5]
```diff
@@ -129,15 +128,6 @@ define([
 
             _fullscreenState = false;
 
-        // Overwrite the event dispatchers to block on certain occasions
-        this.sendEvent = function() {
-            if (!_attached) {
```
Comment (robwalch) [inline src/js/providers/html5.js]: Again, this functionality has been removed. Events should not be triggered if _attached is false.

**Pull request closed without merging**